src/Jackett.Common/Models/ResultPage.cs

```diff
@@ -41,42 +43,42 @@ public string ToXml(Uri selfAtom)
                     new XAttribute(XNamespace.Xmlns + "torznab", torznabNs.NamespaceName),
                     new XElement("channel",
                         new XElement(atomNs + "link",
-                            new XAttribute("href", selfAtom.ToString()),
+                            new XAttribute("href", selfAtom.AbsoluteUri),
                             new XAttribute("rel", "self"),
                             new XAttribute("type", "application/rss+xml")
                         ),
                         new XElement("title", ChannelInfo.Title),
                         new XElement("description", ChannelInfo.Description),
-                        new XElement("link", ChannelInfo.Link),
+                        new XElement("link", ChannelInfo.Link.AbsoluteUri),
                         new XElement("language", ChannelInfo.Language),
                         new XElement("category", ChannelInfo.Category),
                         new XElement("image",
-                            new XElement("url", ChannelInfo.ImageUrl.ToString()),
+                            new XElement("url", ChannelInfo.ImageUrl.AbsoluteUri),
                             new XElement("title", ChannelInfo.ImageTitle),
-                            new XElement("link", ChannelInfo.ImageLink.ToString()),
+                            new XElement("link", ChannelInfo.ImageLink.AbsoluteUri),
                             new XElement("description", ChannelInfo.ImageDescription)
                         ),
                         from r in Releases
                         select new XElement("item",
                             new XElement("title", r.Title),
-                            new XElement("guid", r.Guid),
+                            new XElement("guid", r.Guid.AbsoluteUri),  // GUID and (Link or Magnet) are mandatory
                             new XElement("jackettindexer", new XAttribute("id", r.Origin.ID), r.Origin.DisplayName),
-                            r.Comments == null ? null : new XElement("comments", r.Comments.ToString()),
+                            r.Comments == null ? null : new XElement("comments", r.Comments?.AbsoluteUri),
```

Comment:
we check for null already either remove `r.Comments == null ?` conditional, or change to `r.Comments.AbsoluteUri`

Reply:
Thinking about it, since if comments is null we're making it null, the best choice is
```suggestion
                            new XElement("comments", r.Comments?.AbsoluteUri),
```

Reply:
I changed it to `r.Comments.AbsoluteUri`.
I can't accept this suggestion because I don't want to create the XML element if it's null.